cognite/powerops/resync/v1/config_to_fdm.py

```diff
@@ -0,0 +1,528 @@
+
+                    parsed_properties[property_configuration.property] = [
+                        subtype.external_id for subtype in subtype_list
+                    ]
+                    for subtype in subtype_list:
+                        if subtype.external_id in generated_data_model_objects:
+                            logger.warning(
+                                f"Subtype {subtype.external_id} already exists in generated data model objects"
+                            )
+                        generated_data_model_objects[subtype.external_id] = subtype
+                else:
+                    subtype = self._get_subtype_from_dictionary(property_configuration, raw_data)
+
+                    parsed_properties[property_configuration.property] = subtype.external_id
+                    generated_data_model_objects[subtype.external_id] = subtype
+            else:
+                parsed_properties[property_configuration.property] = self._get_property_value_from_source(
+                    property_configuration, raw_data
+                )
+
+        return parsed_properties, generated_data_model_objects
+
+    # TODO: check if this is the correct way to handle connections
+    def _get_all_connections(self, data: dict, power_asset_type_a: str, power_asset_type_b: str) -> list[str]:
+        """Creates a list of all connections based on the provided data and asset types.
+
+        Provided the data input and asset types the connections will be filtered based on the asset types and the
+        connections will be returned as a list of external ids.
+
+        Args:
+            data: A dictionary of data to be used to filter the connections.
+            power_asset_type_a: A string of the first power asset type to filter the connections.
+            power_asset_type_b: A string of the second power asset type to filter the connections.
+
+        Returns:
+            connections: A list of all connections external ids based on the provided data and asset types.
+        """
+
+        all_connections = self._get_property_value_from_source(
+            PropertyConfiguration(
+                property="connections",
+                source_file=Path("files/model.yaml"),
+                extraction_path="connections",
+                cast_type=None,
+            ),
+            data,
+        )
+        connections = []
+
+        name = data["name"]
+
+        for connection in all_connections:
+            if (
+                connection["from"] == name
+                and connection.get("from_type") == power_asset_type_a
+                and connection.get("to_type") == power_asset_type_b
+            ):
+                connections.append(connection["to"])
+            elif (
+                connection["to"] == name
+                and connection.get("to_type") == power_asset_type_a
+                and connection.get("from_type") == power_asset_type_b
```

Comment:
This depends on from_type and to_type to be set. That's not a requirement from Shop (it is OK to define connections without from_type and to_type, as long as the value/name in "from" and "to", respectively, is unique across types.)
It may be fine to require these on our side to save some complexity, but then we should be explicit about that requirement (somewhere in documentation?)

Reply:
As we have access to the whole yaml file, we could create functions like this:
```
def from_type(connection: dict[str, str], model: dict[str, Any]) -> str:
    if "from_type" in connection:
        return connection["from_type"]
    from_object = connection["from"]
    for object_type, objects in model.items():
        if from_object in objects:
            return object_type
    raise # some exception indicating that from_type cannot be inferred
```

Reply:
The previous resync implementation made this assumption, I add this as a note in the docs as a requirement and to be improved later.